File: scramble_times.py
#!/usr/bin/python
import sys
import os
import random

def load_times(buffer):
    times = []
    prev = '0.000'
    i = 1
    for line in buffer.readlines():
        if line.startswith('t='):
            time = line.split()[0].split('=')[1][:-1]
            times.append('%d: %s-%s\n' % (i, prev, time))
            prev = time
            i = i + 1
    return times


def randomize(times):
    length = len(times) - 1
    for swap in range(length * 5):
        rval1 = random.randint(0, length)
        rval2 = random.randint(0, length)
        times[rval1], times[rval2] = times[rval2], times[rval1]


if __name__ == '__main__':
    if len(sys.argv) < 2:
        print 'usage: scramble_times mp3_file [outfile]'
        sys.exit()

    child_out = os.popen('mpgedit -vv ' + sys.argv[1])
    times = load_times(child_out)
    randomize(times)
    if len(sys.argv) > 2:
        file(sys.argv[2], 'w').writelines(times)
    for time in times:
        print time,

File: scramble_times.pl
#!/usr/bin/perl
sub randomize(@)
{
  my (@tarray) = @_;
  my $len, $cnt, $rval1, $rval2, $tmp;

  $len = @tarray;
  $cnt = $len * 5;
  while ($cnt > 0) {
    $rval1 = int(rand($len));
    $rval2 = int(rand($len));
    $tmp = $tarray[$rval1];
    $tarray[$rval1] = $tarray[$rval2];
    $tarray[$rval2] = $tmp;
    $cnt--;
  }
  return @tarray;
}


sub load_times_array($)
{
  my ($file) = @_;
  my @vvoutput, @times, $prev, $i, $str;

  @vvoutput=`mpgedit -vv "$file"`;
  $prev = "0.000";
  $i=1;
  for (@vvoutput) {
    chop;
    if (/^t=/) {
      s/s.*//;
      s/t=//;
      $str = "$i: ${prev}-$_";
      $prev = "$_";
      push(@times, $str);
      $i++;
    }
  }
  return @times;
}


# --------------- main ------------------
if ($#ARGV == -1) {
  print("usage: $0 mp3_file\n");
  exit(1);
}

$ENV{"PATH"} = ".:" . $ENV{"PATH"};

# Seed random number generator with known value; for
# debugging only!!!
#
#srand(1);

@times = load_times_array("$ARGV[0]");
@times = randomize(@times);
if ($#ARGV > 0) {
  open(FP, ">$ARGV[1]") || die ("failed opening output file '$ARGV[1]'\n");
  for (@times) {
    print(FP "$_\n");
  }
  close(FP);
}
else {
  for (@times) {
    print("$_\n");
  }
}
